sympy/combinatorics/permutations.py

```diff
@@ -81,34 +81,19 @@ def _af_rmuln(*abc):
     """
     a = abc
     m = len(a)
-    if m == 3:
-        p0, p1, p2 = a
-        return [p0[p1[i]] for i in p2]
-    if m == 4:
-        p0, p1, p2, p3 = a
-        return [p0[p1[p2[i]]] for i in p3]
-    if m == 5:
-        p0, p1, p2, p3, p4 = a
-        return [p0[p1[p2[p3[i]]]] for i in p4]
-    if m == 6:
-        p0, p1, p2, p3, p4, p5 = a
-        return [p0[p1[p2[p3[p4[i]]]]] for i in p5]
-    if m == 7:
-        p0, p1, p2, p3, p4, p5, p6 = a
-        return [p0[p1[p2[p3[p4[p5[i]]]]]] for i in p6]
-    if m == 8:
-        p0, p1, p2, p3, p4, p5, p6, p7 = a
-        return [p0[p1[p2[p3[p4[p5[p6[i]]]]]]] for i in p7]
-    if m == 1:
-        return a[0][:]
-    if m == 2:
-        a, b = a
-        return [a[i] for i in b]
+
     if m == 0:
         raise ValueError("String must not be empty")
-    p0 = _af_rmuln(*a[:m//2])
-    p1 = _af_rmuln(*a[m//2:])
-    return [p0[i] for i in p1]
+
+    if m == 1:
+        return a[0][:]
+
+    if m > 2:
+        p = _af_rmuln(*a[1:])
+        return [a[0][i] for i in p]
+
+    elif m > 1:
+        return [a[-2][i] for i in a[-1]]
```

Comment:
Can this just be:
```python
def _af_rmuln2(*permutations):
    result, perms = permutations[0], permutations[1:]
    for p in perms:
        result = [result[i] for i in p]
    return result
```

Reply:
Yes, that's a much better way to write it. Just checked a bunch of cases, and I think it works. Thank you.

Reply:
You can just reverse the order.